### PR TITLE
init: the setup of the typeorm entities and relations

### PR DESCRIPTION
**Pull request by GalOren33**
entities and relation

### CODE REVIEW

File: src/entities/allergies.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	PrimaryGeneratedColumn,
+	ManyToOne,
+	JoinColumn,
+} from "typeorm";
+import { participants } from "./participants";
+@Entity("allergies")
+export class allergies {
+	@PrimaryGeneratedColumn()
+	id: number;
+	@Column({})
```
[the remark this answers is not in the document]
Comment (davidtrainers) [reply src/entities/allergies.ts:13]: add text type

File: src/entities/allergies.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import {
+	Entity,
+	Column,
+	PrimaryGeneratedColumn,
+	ManyToOne,
+	JoinColumn,
+} from "typeorm";
+import { participants } from "./participants";
+@Entity("allergies")
+export class allergies {
```
Comment (davidtrainers) [inline src/entities/allergies.ts:10]: Capital letter Allergies

**Review by davidtrainers — CHANGES_REQUESTED** [review]

File: src/entities/allergies.ts (hunk 2 above)
Comment (GalOren33) [reply src/entities/allergies.ts:10]: done for all

File: src/entities/allergies.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (GalOren33) [reply src/entities/allergies.ts:13]: done

File: src/entities/allergies.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	@PrimaryGeneratedColumn()
+	id: number;
+	@Column({})
+	name: string;
+	@Column({ nullable: false })
```
[the remark this answers is not in the document]
Comment (GalOren33) [reply src/entities/allergies.ts:15]: done

File: src/entities/measurements..entity.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { Participants } from "./participants.entity";
+@Entity("measurements")
+export class Measurements {
+	@PrimaryGeneratedColumn()
+	id: number;
+
+	@Column({ default: false })
+	is_deleted: boolean;
+
+	@Column()
```
Comment (davidtrainers) [inline src/entities/measurements..entity.ts:18]: add numeric type

File: src/entities/measurements..entity.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	height: number;
+
+	@Column()
```
Comment (davidtrainers) [inline src/entities/measurements..entity.ts:21]: add numeric type

File: src/entities/planes.entity.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import {
```
Comment (davidtrainers) [inline src/entities/planes.entity.ts:1]: change file name to plans

File: src/entities/planes.entity.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+	@Column({ type: "timestamp with time zone" })
+	start_basic: Timestamp;
+
+	@Column({ type: "timestamp with time zone" })
+	start_advnaced: Timestamp;
+
+	@Column({ type: "timestamp with time zone" })
+	end_date: Timestamp;
+
+	@Column()
```
Comment (davidtrainers) [inline src/entities/planes.entity.ts:23]: Where is the reference to the forces table

File: src/entities/results.entity.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+	@Column({ nullable: false })
+	event_type_comp_id: number;
+
+	@ManyToOne(() => Participants, (participant) => participant.results)
+	@JoinColumn()
+	participant_id: Participants;
+
+	@ManyToOne(() => Components, (component) => component.results)
+	@JoinColumn()
+	component: Components[];
```
Comment (davidtrainers) [inline src/entities/results.entity.ts:35]: Why component and also event_type_comp_id

File: src/entities/scores.entity.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+import { EventTypeCompCon } from "./event_type_comp_con.entity";
+@Entity("scores")
+export class Scores {
+	@PrimaryGeneratedColumn()
+	id: number;
+
+	@Column({ nullable: false })
+	score: number;
+	
+	@Column()
```
Comment (davidtrainers) [inline src/entities/scores.entity.ts:18]: add numeric type

File: src/entities/scores.entity.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+	ManyToOne,
+	JoinColumn,
+} from "typeorm";
+import { EventTypeCompCon } from "./event_type_comp_con.entity";
+@Entity("scores")
+export class Scores {
+	@PrimaryGeneratedColumn()
+	id: number;
+
+	@Column({ nullable: false })
```
Comment (davidtrainers) [inline src/entities/scores.entity.ts:15]: add numeric type

File: src/entities/weeks.entity.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+export class Weeks {
+	@PrimaryGeneratedColumn()
+	id: number;
+
+	@Column({ default: false })
+	is_deleted: boolean;
+
+	@Column()
+	title: number;
+
```
Comment (davidtrainers) [inline src/entities/weeks.entity.ts:22]: Change to string and add text type

**Review by davidtrainers — CHANGES_REQUESTED** [review]

File: src/entities/allergies.entity.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+	
+	@Column({type: "text"})
+	name: string;
+	
+	@Column({ nullable: false })
+	is_deleted: boolean;
+
+	@ManyToOne(() => Participants, (participant) => participant.allergies)
+	@JoinColumn()
+	participant_id: Participants;
```
Comment (davidtrainers) [inline src/entities/allergies.entity.ts:22]: remove id

File: src/entities/categories.entity.ts [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+	@Column({ default: 1,type:"numeric" })
+	num_in_seq: number;
+
+	@OneToMany((type) => EventType, (event_type) => event_type.event_category_id)
+	event_type: EventType[];
+	
+	@ManyToOne((type) => Categories, (category) => category.children)
+	@JoinColumn()
+	parent_id: Categories;
```
Comment (davidtrainers) [inline src/entities/categories.entity.ts:30]: remove id

File: src/entities/event_type.entity.ts [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+	@Column({ default: false })
+	is_intense: boolean;
+	
+	@ManyToOne(() => Instructions, (instruction) => instruction.event_types)
+	@JoinColumn()
+	instruction: Instructions;
+
+	@ManyToOne(() => Categories, (category) => category.event_type)
+	@JoinColumn()
+	event_category_id: EventType;
```
Comment (davidtrainers) [inline src/entities/event_type.entity.ts:35]: remove id

File: src/entities/event_type_comp_con.entity.ts [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+	@OneToMany((type) => Scores, (score) => score.event_type_comp_con)
+	scores: Scores[];
+
+	@ManyToOne((type) => Components, (component) => component.event_types)
+	@JoinColumn()
+	
+	event_component_id: Components;
+	@ManyToOne((type) => EventType, (event_type) => event_type.components)
+	@JoinColumn()
+	event_type_id: Components;
```
Comment (davidtrainers) [inline src/entities/event_type_comp_con.entity.ts:25]: remove id

File: src/entities/events.entity.ts [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+	@Column({ type: "timestamp with time zone" })
+	event_date: Timestamp;
+
+	@ManyToOne((type) => Forces, (force) => force.events)
+	@JoinColumn()
+	force_id: Forces;
+
+	@ManyToOne((type) => Weeks, (week) => week.events)
+	@JoinColumn()
+	week_id: Weeks;
```
Comment (davidtrainers) [inline src/entities/events.entity.ts:37]: remove id

File: src/entities/events.entity.ts [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+
+	@Column({ type: "timestamp with time zone" })
+	created_date: Timestamp;
+	
+	@Column({ type: "timestamp with time zone" })
+	event_date: Timestamp;
+
+	@ManyToOne((type) => Forces, (force) => force.events)
+	@JoinColumn()
+	force_id: Forces;
```
Comment (davidtrainers) [inline src/entities/events.entity.ts:33]: remove id

File: src/entities/forces.entity.ts [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+
+	@Column("text")
+	name: string;
+
+	@OneToMany((type) => Users, (user) => user.force)
+	users: Users[];
+
+	@ManyToOne((type) => Forces, (force) => force.children)
+	@JoinColumn()
+	parent_id: Forces;
```
Comment (davidtrainers) [inline src/entities/forces.entity.ts:29]: remove id

File: src/entities/measurements.entity.ts [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+
+	@Column({type: 'numeric'})
+	weight: number;
+
+	@Column({ type: "timestamp with time zone" })
+	date: Date;
+	
+	@ManyToOne((type) => Participants, (participant) => participant.measurements)
+	@JoinColumn()
+	participant_id: Participants;
```
Comment (davidtrainers) [inline src/entities/measurements.entity.ts:28]: remove id

File: src/entities/participants.entity.ts [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+export class Participants {
+	@PrimaryGeneratedColumn()
+	id: number;
+	
+	@Column({ default: false })
+	is_deleted: boolean;
+
+	@OneToOne((type) => Users, (user) => user.participant)
+	@JoinColumn()
+	user_id: Users;
```
Comment (davidtrainers) [inline src/entities/participants.entity.ts:24]: remove id

File: src/entities/restrections.entity.ts [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+
+	@Column({ type: "timestamp with time zone" })
+	start_date: Timestamp;
+
+	@Column({ type: "timestamp with time zone" })
+	end_date: Timestamp;
+
+	@ManyToOne(() => Participants, (participant) => participant.restrections)
+	@JoinColumn()
+	participant_id: Participants;
```
Comment (davidtrainers) [inline src/entities/restrections.entity.ts:30]: remove id

File: src/utils/logger.ts [hunk 22]
```diff
@@ -36,7 +36,7 @@ const logger = winston.createLogger({
       json: false,
       zippedArchive: true,
     }),
-    // error log setting
+    // // error log setting
```
Comment (davidtrainers) [inline src/utils/logger.ts:39]: remove

File: src/routes/index.route.ts [hunk 23]
```diff
@@ -1,6 +1,9 @@
 import { Router } from 'express';
 import IndexController from '@controllers/index.controller';
 import { Routes } from '@interfaces/routes.interface';
+import { createConnection, getRepository } from 'typeorm';
```
Comment (davidtrainers) [inline src/routes/index.route.ts:4]: remove three imports

File: src/routes/events.route.ts [hunk 24]
```diff
@@ -1,19 +1,24 @@
...
+import { Forces } from '@/entities/forces.entity';
 // import validationMiddleware from '@middlewares/validation.middleware';
 
 class EventsRoute implements Routes {
-    public path = '/events';
-    public router = Router();
-    //   public usersController = new eventsController();
+  public path = '/events';
+  public router = Router();
+  //   public usersController = new eventsController();
```
Comment (davidtrainers) [inline src/routes/events.route.ts:10]: remove

File: src/entities/allergies.entity.ts [hunk 25]
```diff
@@ -0,0 +1,23 @@
...
+import { Participants } from "./participants.entity";
+@Entity("allergies")
+export class Allergies {
+	@PrimaryGeneratedColumn()
+	id: number;
+	
+	@Column({type: "text"})
+	name: string;
+	
+	@Column({ nullable: false })
```
Comment (davidtrainers) [inline src/entities/allergies.entity.ts:17]: add default

**Review by davidtrainers — CHANGES_REQUESTED** [review]

**Review by davidtrainers — APPROVED** [review]

**Pull request merged**